core/src/components/datetime/datetime.tsx

```diff
@@ -912,6 +867,29 @@ export class Datetime implements ComponentInterface {
         const monthBox = month.getBoundingClientRect();
         if (Math.abs(monthBox.x - box.x) > 2) return;
 
+        /**
+         * If we're force-rendering a month, and we've scrolled to
+         * that month, return that.
+         *
+         * Checking that we've actually scrolled to the forced month
+         * is mostly for safety; in theory, if there's a forced month,
+         * that means a new value was manually set, so we should have
+         * automatically animated directly to it.
+         */
+        const { forceRenderDate } = this;
+        const firstDayEl = month.querySelector('.calendar-day');
+        const dataMonth = firstDayEl?.getAttribute('data-month');
+        const dataYear = firstDayEl?.getAttribute('data-year');
+        if (
+          forceRenderDate !== null &&
+          dataMonth &&
+          dataYear &&
+          parseInt(dataMonth) === forceRenderDate.month &&
+          parseInt(dataYear) === forceRenderDate.year
+        ) {
+          return { month: forceRenderDate.month, year: forceRenderDate.year, day: forceRenderDate.day };
```

Comment:
What should datetime do when being set to an out of bounds value? Right now it scrolls as far as it can:

```html
<ion-datetime max="2023"></ion-datetime>
```

```js
const datetime = document.querySelector('ion-datetime');

setTimeout(() => {
  datetime.value = "2024-01-01T16:30"
}, 2000);
```

However, this brings me to December 2023 which is confusing since I set the datetime to January 2024.